packages/scripts/src/migration/dvm2-designated-voting/0_Deploy.ts

```diff
@@ -0,0 +1,113 @@
+const hre = require("hardhat");
+const fetch = require("node-fetch");
+
+import {
+  VotingTokenEthers,
+  DesignatedVotingFactoryEthers,
+  DesignatedVotingEthers,
+  DesignatedVotingV2FactoryEthers,
+} from "@uma/contracts-node";
+import { getContractInstance } from "../../utils/contracts";
+
+const { getContractFactory } = hre.ethers;
+
+async function main() {
+  console.log("Running DesignatedVotingV2 deployments🔥");
+
+  const networkId = Number(await hre.getChainId());
+  if (networkId != 1) throw new Error("Can only run on mainnet");
+  if (!process.env.ETHERSCAN_API_KEY) throw new Error("No ETHERSCAN_API_KEY");
+  if (!process.env.OWNER_TO_MIGRATE) throw new Error("No OWNER_TO_MIGRATE set");
+  const owner = process.env.OWNER_TO_MIGRATE || "";
+
+  const factoryV1 = await getContractInstance<DesignatedVotingFactoryEthers>("DesignatedVotingFactory");
+  const factoryV2 = await getContractInstance<DesignatedVotingV2FactoryEthers>("DesignatedVotingV2Factory");
+  const votingToken = await getContractInstance<VotingTokenEthers>("VotingToken");
+  // Step 1: find all DesignatedVoting contracts created by the DesignatedVotingFactory.
+  const designatedVotingContracts = await _fetchDesignatedVotingContractsCreatedByFactory(factoryV1);
+
+  // Step 2: remove all DesignatedVoting contracts that have 0 UMA.
+  const contractsTokenBalances = await Promise.all(
+    designatedVotingContracts.map(async (address) => votingToken.balanceOf(address))
+  );
+  const designatedVotingContractsWithTokens: DesignatedVotingEthers[] = [];
+  for (let i = 0; i < designatedVotingContracts.length; i++)
```

Comment:
Similar comment as below on these loops. Not necessary to change, but maps and filters abstract away the concept of an index so ts has certainty that an element exists. You also lose concurrency benefits when using typical for loops with awaits inside.

Reply:
ye, I specifically wanted to lose the concurrency benefits to ensure that all contract instance constructions are done (the `await getContractInstance...`) before continuing. you are right that there is a cleaner pattern for sure but I'll leave it like this as it's really easy to understand.